src/commands/pull.js

```diff
@@ -0,0 +1,107 @@
+const { getLoggedClientOrThrow } = require('../lib/utils');
+
+class PullCommand extends ApifyCommand {
+    async run() {
+        const { args } = this.parse(PullCommand);
+        const apifyClient = await getLoggedClientOrThrow();
+        const cwd = process.cwd();
+
+        const { actorId } = args;
+        const actor = await apifyClient.actor(actorId).get();
+        if (!actor) throw new Error(`Cannot find actor with ID '${actorId}' in your account.`);
+        const { name, versions } = actor;
+
+        const latestVersion = versions.reduce((match, curr) => {
+            if (semverGt(`${curr.versionNumber}.0`, `${match.versionNumber}.0`)) return curr;
+            return match;
+        });
+
+        const dirpath = path.join(cwd, name);
+        fs.mkdirSync(dirpath, { recursive: true }, null);
+
+        if (!(fs.readdirSync(dirpath).length === 0)) {
+            error(`Directory ${dirpath} is not empty. Please empty it or choose another directory.`);
+        }
+
+        let isPullSuccessful = false;
+
+        switch (latestVersion.sourceType) {
+            case 'TARBALL': {
+                const { tarballUrl } = latestVersion;
+
+                const { data } = await get(tarballUrl, { responseType: 'arraybuffer' });
+
+                const zipFile = new AdmZip(Buffer.from(data, 'binary'));
+                zipFile.extractAllTo(dirpath);
+                isPullSuccessful = true;
+                break;
+            }
+            case 'SOURCE_FILES': {
+                const { sourceFiles } = latestVersion;
+                for (const file of sourceFiles) {
+                    if (!file.folder) {
+                        const fileParts = file.name.split('/');
+
+                        if (fileParts.length > 1) {
+                            const folderPath = fileParts.slice(0, -1).join('/');
+                            fs.mkdirSync(`${dirpath}/${folderPath}`, { recursive: true }, null);
+                        }
+                        const fileContent = file.format === 'BASE64' ? Buffer.from(file.content, 'base64') : file.content;
+                        fs.writeFileSync(`${dirpath}/${file.name}`, fileContent);
+                    }
+                }
+                isPullSuccessful = true;
+                break;
+            }
+            case 'GIT_REPO': {
+                // e.g. https://github.com/jakubbalada/Datasety.git#master:RejstrikPolitickychStran
+                const { gitRepoUrl } = latestVersion;
+                const [repoUrl, branchDirPart] = gitRepoUrl.split('#');
+
+                let branch;
+                let dir;
+                if (branchDirPart) [branch, dir] = branchDirPart.split(':');
+                let branchDirRepoUrl = repoUrl;
+                if (dir) branchDirRepoUrl += `/${dir}`;
+                if (branch) branchDirRepoUrl += `#${branch}`;
+
+                const emitter = tiged(branchDirPart ? branchDirRepoUrl : repoUrl);
```

Comment:
This doesn't need the ternary anymore, just `tiged(branchDirPart)` should do the same thing

Reply:
It wouldn't use selected branch and directory in that case.

Reply:
Sorry, I meant `tiged(branchDirRepoUrl)`